aio/src/app/custom-elements/events/events.component.spec.ts

```diff
@@ -31,162 +29,267 @@ describe('EventsComponent', () => {
-      eventsService.events.next(testEvents);
+      eventsService.events.next([{
+        'location': 'Oslo, Norway',
+        'tooltip': 'ngVikings',
+        'linkUrl': 'https://ngvikings.org/',
+        'name': 'ngVikings',
+        'date': {
+          'start': '2099-07-25',
+          'end': '2099-07-26'
+        },
+        'workshopsDate': {
+          'start': '2099-07-27',
+          'end': '2099-07-27'
+        }
+      },
+      {
+        'location': 'Salt Lake City, Utah',
+        'tooltip': 'ng-conf',
+        'linkUrl': 'https://ng-conf.org/',
+        'name': 'ng-conf  ',
+        'date': {
+          'start': '2099-06-20',
+          'end': '2099-06-20'
+        }
+      },
+      {
+        'location': 'Delhi, India',
+        'tooltip': 'ngIndia',
+        'linkUrl': 'https://www.ng-ind.com/',
+        'name': 'ngIndia',
+        'date': {
+          'start': '2020-02-29',
+          'end': '2020-02-29'
+        }
+      }]);
       expect(component.upcomingEvents.length).toEqual(2);
     });
 
-    it('should have 5 past events', () => {
-      eventsService.events.next(testEvents);
-      expect(component.pastEvents.length).toEqual(6);
+    it('should have 4 past events', () => {
+      eventsService.events.next([
+      {
+        'location': 'Salt Lake City, Utah',
+        'tooltip': 'ng-conf',
+        'linkUrl': 'https://ng-conf.org/',
+        'name': 'ng-conf  ',
+        'date': {
+          'start': '2099-06-20',
+          'end': '2099-06-20'
+        }
+      },
+      {
+        'location': 'Rome, Italy',
+        'tooltip': 'NG Rome MMXIX - The Italian Angular Conference',
+        'linkUrl': 'https://ngrome.io',
+        'name': 'NG Rome MMXIX',
+        'date': {
+          'start': '2019-10-07',
+          'end': '2019-10-07'
+        },
+        'workshopsDate': {
+          'start': '2019-10-06',
+          'end': '2019-10-06'
+        }
+      },
+      {
+        'location': 'London, UK',
+        'tooltip': 'AngularConnect',
+        'linkUrl': 'https://www.angularconnect.com/?utm_source=angular.io&utm_medium=referral',
+        'name': 'AngularConnect',
+        'date': {
+          'start': '2019-09-19',
+          'end': '2019-09-20'
+        }
+      },
+      {
+        'location': 'Berlin, Germany',
+        'tooltip': 'NG-DE',
+        'linkUrl': 'https://ng-de.org/',
+        'name': 'NG-DE',
+        'date': {
+          'start': '2019-08-30',
+          'end': '2019-08-31'
+        },
+        'workshopsDate': {
+          'start': '2019-08-29',
+          'end': '2019-08-29'
+        }
+      },
+      {
+        'location': 'Oslo, Norway',
+        'tooltip': 'ngVikings',
+        'linkUrl': 'https://ngvikings.org/',
+        'name': 'ngVikings',
+        'date': {
+          'start': '2018-07-26',
+          'end': '2018-07-26'
+        },
+        'workshopsDate': {
+          'start': '2018-07-27',
+          'end': '2018-07-27'
+        }
+      }]);
```

Comment:
```suggestion
        }
      ]);
```

(And indent the rest of the items accordingly.)